dataikuapi/dss/ml.py

```diff
@@ -1791,6 +1981,235 @@ def get_algorithm_settings(self, algorithm_name):
+        time series resampling parameters, not a copy, so changes made to the returned object will be reflected when saving
+
+        :return: A dict of the resampling parameters
+        :rtype: dict
+        """
+        return self.mltask_settings["preprocessing"]["timeseriesSampling"]
+
+    def set_numerical_interpolation(self, method=None, constant=None):
+        """
+        Sets the time series resampling numerical interpolation parameters
+
+        :param method: Interpolation method. Valid values are: NEAREST, PREVIOUS, NEXT, LINEAR, QUADRATIC, CUBIC, CONSTANT
+        :type method: str
+        :param constant: Value for the CONSTANT interpolation method
+        :type constant: float
+        :return:
+        """
+        resampling_params = self.get_resampling_params()
+        if method is not None:
+            assert method in self.INTERPOLATION_METHODS, "Invalid interpolation method, should be in {}".format(self.INTERPOLATION_METHODS)
+            resampling_params["numericalInterpolateMethod"] = method
+        if constant is not None:
+            resampling_params["numericalInterpolateConstantValue"] = constant
+
+    def set_numerical_extrapolation(self, method=None, constant=None):
+        """
+        Sets the time series resampling numerical extrapolation parameters
+
+        :param method: Extrapolation method. Valid values are: PREVIOUS_NEXT, NO_EXTRAPOLATION, CONSTANT, LINEAR, QUADRATIC, CUBIC
+        :type method: str
+        :param constant: Value for the CONSTANT extrapolation method
+        :type constant: float
+        :return:
+        """
+        resampling_params = self.get_resampling_params()
+        if method is not None:
+            assert method in self.EXTRAPOLATION_METHODS, "Invalid extrapolation method, should be in {}".format(self.EXTRAPOLATION_METHODS)
+            resampling_params["numericalExtrapolateMethod"] = method
+        if constant is not None:
+            resampling_params["numericalExtrapolateConstantValue"] = constant
+
+    def set_categorical_imputation(self, method=None, constant=None):
+        """
+        Sets the time series resampling categorical imputation parameters
+
+        :param method: Imputation method. Valid values are: MOST_COMMON, NULL, CONSTANT, PREVIOUS_NEXT, PREVIOUS, NEXT
+        :type method: str
+        :param constant: Value for the CONSTANT imputation method
+        :type constant: str
+        :return:
+        """
+        resampling_params = self.get_resampling_params()
+        if method is not None:
+            assert method in self.CATEGORICAL_IMPUTATION_METHODS, "Invalid imputation method, should be in {}".format(self.CATEGORICAL_IMPUTATION_METHODS)
+            resampling_params["categoricalImputeMethod"] = method
+        if constant is not None:
+            resampling_params["categoricalConstantValue"] = constant
+
+    def set_duplicate_timestamp_handling_method(self, method):
```

Comment:
More future-proof if we ever add other params + more consistent with other params

```suggestion
    def set_duplicate_timestamp_handling(self, method):
```